lib/marklogic.js

```diff
@@ -369,6 +371,42 @@ function MarkLogicClient(connectionParams) {
 
   this.logger = null;
 }
+
+/**
+ * Tests if a connection is successful.
+ * @method DatabaseClient#testConnection
+ * @since 2.1
+ * @returns true if the connection is successful and false otherwise with an error message
+ */
+MarkLogicClient.prototype.testConnection = function testConnection() {
+  var connectionParams = this.connectionParams;
+  var requestOptions = mlutil.copyProperties(connectionParams);
+  requestOptions.method = 'HEAD';
+  requestOptions.path = mlutil.databaseParam(connectionParams, '/v1/ping', '?');
+  var operation = new Operation(
+      'test operation', this, requestOptions, 'empty', 'single'
+      );
+  operation.statusCodeValidator = function testStatusCodes(statusCode, response){
+    if (statusCode >= 300)
+        this.responseStatusMessage = response.statusMessage;
+    return null;
+   };
+  operation.outputTransform = function addOutputTransform() {
+    var content = null;
+    if(this.responseStatusCode <300) {
+       content = {connected: true};
+    } else {
+       content = {connected: false,
+       httpStatusCode:  this.responseStatusCode,
+       httpStatusMessage: operation.responseStatusMessage
```

Comment:
Should the assignment use this.responseStatusMessage instead of operation.responseStatusMessage for consistency and to avoid side effects if the code is refactored to move the addOutputTransform() function out of the the testConnection() function in the future?

The this object refers to the object on which the function is called while operation is a closure on the local variable of testConnection()